Added button to cart page

metehansalih06 opened the pull request.

metehansalih06 committed:
Added button to cart page

ookii-tsuki commented:
Can you add a remove item from cart button? like an X button on every cart Item on something. I forgot to add it to the docs but it's implemented in the server. you just have to call `POST /api/order/remove_cart` with:
```json
{
    item_id: "1"
}
```

metehansalih06 committed:
Added remove button

ookii-tsuki reviewed (commented):
- The webpage shows that there's an error when the user is not logged in
> Fehler beim Abrufen des Warenkorbs. Bitte versuchen Sie es später erneut.

instead of a message telling them they're not logged in.

- The remove button does nothing

metehansalih06 commented:
,,instead of a message telling them they're not logged in''. I can deal with that later if there is time. The checkout page makes me problems so I am busy trying to fix the different issues. I am not able to test by myself as I cant add cart items to the db. Do you get an error when you try to delete the item from the cart or does it do nothing at all

ookii-tsuki commented:
When you try to delete the an item from cart you get an error from the server `404` and that's because you're sending `"undefined"` instead of an id. the problematic line appears to be this one at line 313:
```js
const itemId = this.getAttribute('data-item-id');
```

metehansalih06 committed:
Changed remove button

I changed the line for the item id which sent undefined. I added the functionality to change the amount of an item by clicking on the space on the card which shows the quantity. The required number can be typed in.

ookii-tsuki commented:
It's broken all together now. when you have an item in the cart, it will give me an error and not load anything. the error is:
```
cart.js:283 Fehler beim Abrufen des Warenkorbs. ReferenceError: cartItemDiv is not defined
    at displayCart (cart.js:333:5)
    at fetchCart (cart.js:281:6)
```

metehansalih06 commented:
I will try to fix that tomorrow. However I can remove the new functionality with the amount and just push the change of X-button. Maybe the issue is with the new code.

metehansalih06 committed:
Removed the quantity selection

metehansalih06 committed:
Merge branch 'main' into Metehan-cart-page

metehansalih06 committed:
The remove button works now

metehansalih06 committed:
cart page is finished

Everything works now. The user can remove an item from the cart, change the quantity and add a note.

ookii-tsuki reviewed (approved):
works
